compiler/src/dotty/tools/scripting/Util.scala

```diff
@@ -0,0 +1,60 @@
+package dotty.tools.scripting
+
+import java.nio.file.{ Path }
+import java.io.File
+import java.net.{ URLClassLoader }
+import java.lang.reflect.{ Modifier, Method }
+
+object Util:
+
+  def deleteFile(target: File): Unit =
+    if target.isDirectory then
+      for member <- target.listFiles.toList
+      do deleteFile(member)
+    target.delete()
+  end deleteFile
+
+  def detectMainClassAndMethod(outDir: Path, classpathEntries: Seq[Path], srcFile: String): (String, Method) =
+    val classpathUrls = (classpathEntries :+ outDir).map { _.toUri.toURL }
+    val cl = URLClassLoader(classpathUrls.toArray)
+
+    def collectMainMethods(target: File, path: String): List[(String, Method)] =
+      val nameWithoutExtension = target.getName.takeWhile(_ != '.')
+      val targetPath =
+        if path.nonEmpty then s"${path}.${nameWithoutExtension}"
+        else nameWithoutExtension
+
+      if target.isDirectory then
+        for
+          packageMember <- target.listFiles.toList
+          membersMainMethod <- collectMainMethods(packageMember, targetPath)
+        yield membersMainMethod
+      else if target.getName.endsWith(".class") then
+        val cls = cl.loadClass(targetPath)
+        try
+          val method = cls.getMethod("main", classOf[Array[String]])
+          if Modifier.isStatic(method.getModifiers) then List((cls.getName, method)) else Nil
+        catch
+          case _: java.lang.NoSuchMethodException => Nil
+      else Nil
+    end collectMainMethods
+
+    val mains = for
+      file <- outDir.toFile.listFiles.toList
+      method <- collectMainMethods(file, "")
+    yield method
+
+    mains match
+      case Nil =>
+        throw StringDriverException(s"No main methods detected for [${srcFile}]")
+      case _ :: _ :: _ =>
+        throw StringDriverException(
+          s"internal error: Detected the following main methods:\n${mains.mkString("\n")}")
+      case m :: Nil => m
+    end match
+  end detectMainClassAndMethod
+
+  def pathsep = sys.props("path.separator")
```

Comment:
There was a previous effort to avoid dependencies on packages that might become modules in 2.13, which did not happen. I'm conflicted about `sys.props` because OT1H it's just a `Map` interface for `System.getProperty` or `ies`, but OTOH it is not so trivial that it didn't harbor a couple of bugs.

I guess the convenience here is `java.io.File.pathSeparator`.  No big deal, I just happened to have recently seen the Scala 2 commit about not using `sys.props`.

Reply:
I wasn't aware of the discussion around packages that might become modules, but  `java.io.File.pathSeparator` is a good alternative in any case.

It turns out that `pathsep` is defined identically in [dotty.tools.scripting.Main](https://github.com/lampepfl/dotty/blob/c4494f9f6d1db23ca2ed8278a957c56c1c87a1b5/compiler/src/dotty/tools/scripting/Main.scala), so the definition maybe belongs in [dotty.tools.scripting.Util](https://github.com/lampepfl/dotty/blob/c4494f9f6d1db23ca2ed8278a957c56c1c87a1b5/compiler/src/dotty/tools/scripting/Util.scala).